### ISSUE TITLE
Crf1d loss does not work

### ISSUE THREAD

**Issue by createmomo** [chainer/chainer#3028]
Hi @beam2d and anyone who may know this problem,

I implemented the LSTM-CRF model of this paper (https://arxiv.org/abs/1603.01360) by chainer.

But when I compared the predicted results with the outputs of original code by the paper author, there were some drastic differences.

For example,
Original code:
word1 B-PERSON
word2 I-PERSON
word3 O

However, although I added the CRF layer, the results below were very common (even if the model had been trained with a very long time)
Chainer:
word1 O
word2 I-PERSON
word3 I-ORG
This results should not have appeared if we add a CRF layer, but in chainer, it is very common. Moreover, the performance is about 65% F-score by the chainer version, but 75% of the original code.

I have checked and compared every where of my code with the original code, the only difference is about the CRF1d loss. 

I am wondering, maybe the problem is the crf1d loss function implemented in chainer?

Many thanks,
MoMo

**Comment by unnonouno** [comment]
Hi, thank you for using CRF feature in Chainer.
You can find an implementation of Bi-LSTM-CNN-CRF:
https://github.com/aonotas/deep-crf
It reported state-of-the-art performance. It may help you.

**Comment by MahdiKalayeh** [comment]
@unnonouno I am looking for dense (fully connected) CRF for semantic segmentation. Do you know if there exist any chainer implementation of it?

**Comment by unnonouno** [comment]
You mean that all variables has relations, right? Chainer's `crf1d` is so called Lafferty's linear-chain CRF: http://repository.upenn.edu/cgi/viewcontent.cgi?article=1162&context=cis_papers It is used in NLP field and is difficult to apply image processing. And I'm sorry I don't know implementation of CRFs for semantic segmentation with Chainer.

**Comment by unnonouno** [comment]
Please refer deep-crf implementation. I'll close this issue. If you can't solve the problem, please reopen it.
